ci-operator/step-registry/cucushift/ota/preupgrade/cucushift-ota-preupgrade-commands.sh

```diff
@@ -239,6 +239,61 @@ function pre-OCP-53921(){
     return 0
 }
 
+function pre-OCP-53907(){
+    echo "Test Start: ${FUNCNAME[0]}"
+    local version
+    version="$(oc get clusterversion --no-headers | awk '{print $2}')"
+    if [ -z "${version}" ] ; then
+        echo "Fail to get cluster version!"
+        return 1
+    fi
+    x_ver=$( echo "${version}" | cut -f1 -d. )
+    y_ver=$( echo "${version}" | cut -f2 -d. )
+    y_ver=$((y_ver+1))
+    ver="${x_ver}.${y_ver}"
+    if ! oc adm upgrade channel candidate-${ver}; then
+        echo "Fail to change channel to candidate-${ver}!"
+        return 1
+    fi
+    local retry=3
+    while (( retry > 0 ));do
+        recommends=$(oc get clusterversion version -o json|jq -r '.status.availableUpdates')
+        if [[ "${recommends}" == "null" ]]; then
+            (( retry -= 1 ))
+            sleep 60
+            echo "No recommended update available! Retry..."
+        else
+            echo "Recommencded update: ${recommends}"
+            break
+        fi
+    done
+    if (( retry == 0 )); then
+        echo "Timeout to get recommended update!" 
+        return 1
+    fi
+    mapfile -t images < <(echo ${recommends}|jq -r '.[].image')
+    if [ -z "${images[*]}" ]; then
+        echo "No image extracted from recommended update!"
+        return 1
+    fi
+    for image in ${images[*]}; do
+        if [[ "${image}" == "null" ]] ; then
+            echo "No image info!"
+            return 1
+        fi
+        metadata=$(oc adm release info ${image} -ojson|jq .metadata.metadata)
```

Comment:
can we directly get `.release.openshift.io/architecture` here to be more accurate?

Reply:
yes, we can. but seems not here, but add one more line after the judgement of metadata=null.

Reply:
```
a=$(oc adm release info quay.io/openshift-release-dev/ocp-release:4.15.2-x86_64 -ojson|jq -r '.metadata.metadata."release.openshift.io/architecture"'); echo $a
null
```
Once that, we can just need one condition to determinate the testing failed or not.

Reply:
It can not distinguish if it's a null metadata or a null architecture, right? when metadata=null, it should continue to next one.

```
# ./oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:0ed362d46466541178ad24c849e9aeacb0c622b5a2cab85cba820330e01a29a1 -ojson|jq -r '.metadata.metadata."release.openshift.io/architecture"'
null
# ./oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:0ed362d46466541178ad24c849e9aeacb0c622b5a2cab85cba820330e01a29a1 -ojson|jq .metadata
{
  "kind": "cincinnati-metadata-v0",
  "version": "4.16.0-ec.1",
  "previous": [
    "4.15.0-ec.0",
    "4.15.0-ec.1",
    "4.15.0-ec.2",
    "4.15.0-ec.3",
    "4.15.0-rc.0",
    "4.15.0-rc.1",
    "4.16.0-ec.0"
  ]
}
```

Reply:
updated the 1st one in [d585db5](https://github.com/openshift/release/pull/49743/commits/d585db51ead597b2083f1ac79b5a4b22346d47cf)

Reply:
--to-latest will pick the latest one in available update and don't care about arch. But cincinnati should send correct arch back when cvo request this info with arch, so there is pre-upgrade step to check if all update in available update has correct arch

Reply:
From my understanding, we can rely on cincinnati returning a correct arch of payload image, but run post-check to ensure the upgrade get successful upgrade, but not have to check each returned image from cincinnati, after in the end, we are expecting cvo and cincinnati should work together correctly, and get the cluster upgraded.

Reply:
> rely on cincinnati returning a correct arch of payload image

Yes, there are two checkpoints here. One of the checkpoint is to check if Cincinnati return a correct arch.  And then pick one to upgrade to check if upgrade succeed. --tolatest is picking one of the available update to upgrade, even upgrade to A succeed, it could not guide all other available updates with correct arch.